LayoutTests/imported/w3c/web-platform-tests/html/cross-origin-opener-policy/resources/noopener-helper.js

```diff
@@ -0,0 +1,159 @@
+      await receive('${popup_reply_token}');
+      const payload = {
+        openee: !!window.openee,
+        closed: window.openee.closed
+      };
+      send('${reply_token}', JSON.stringify(payload));
+    `);
+    t.add_cleanup(() => send(popup_token, 'window.openee && window.openee.close()'));
+    // Notify the popup that its openee was loaded.
+    send(popup_openee_token, `send('${popup_reply_token}', 'popup openee opened');`);
+
+    // Assert that the popup has access to its openee.
+    payload = JSON.parse(await receive(reply_token));
+    assert_true(payload.openee, 'popup openee');
+
+    // Assert that the openee has access to its popup opener.
+    send(popup_openee_token, `
+      let openerDOMAccessAllowed = false;
+      try {
+        openerDOMAccessAllowed = !!self.opener.document.URL;
+      } catch(ex) {
+      }
+      const payload = {
+        opener: !!self.opener,
+        openerDOMAccess: openerDOMAccessAllowed
+      };
+      send('${reply_token}', JSON.stringify(payload));
+    `);
+    payload = JSON.parse(await receive(reply_token));
+    if (opener_expectation) {
+      assert_true(payload.opener, 'Opener is not null');
+      assert_true(payload.openerDOMAccess, 'No DOM access');
+    } else {
+      assert_false(payload.opener, 'Opener is null');
+      assert_false(payload.openerDOMAccess, 'No DOM access');
+    }
+  },
+  'noopener-allow-popups ensures that the opener cannot script the openee,' +
+      ' but further popups with no COOP can access their opener: ' +
+      opener_coop + '/' + openee_coop);
+};
+
+const test_noopener_navigating_away = (popup_coop) => {
+  promise_test(
+      async t => {
+        // Set up dispatcher communications.
+        const popup_token = token();
+        const reply_token = token();
+        const popup_reply_token = token();
+        const popup_second_token = token();
+
+        const popup_url =
+            getExecutorPath(popup_token, SAME_ORIGIN, coop_header(popup_coop));
+
+        // We open a popup and then ping it, it will respond after loading.
+        const popup = window.open(popup_url);
+        send(popup_token, `send('${reply_token}', 'Popup loaded');`);
+        assert_equals(await receive(reply_token), 'Popup loaded');
+        t.add_cleanup(() => send(popup_token, 'window.close()'));
+
+        // Assert that we can script the popup.
+        assert_not_equals(popup.window, null);
+        assert_false(popup.closed, 'popup closed');
+
+        // Ensure that the popup has no access to its opener.
+        send(popup_token, `
+      let openerDOMAccessAllowed = false;
+      try {
+        openerDOMAccessAllowed = !!self.opener.document.URL;
+      } catch(ex) {
+      }
+      const payload = {
+        opener: !!self.opener,
+        openerDOMAccess: openerDOMAccessAllowed
+      };
+      send('${reply_token}', JSON.stringify(payload));
+    `);
+        let payload = JSON.parse(await receive(reply_token));
```

Comment:
nit: indentation
Same a few times in this file

Reply:
done